ts/navigation/MainNavigator.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from "react";
+import ROUTES from "./routes";
+
+import { Icon } from "native-base";
+import { TabBarBottom, TabNavigator } from "react-navigation";
+import MessagesScreen from "../screens/main/MessagesScreen";
+import ProfileScreen from "../screens/main/ProfileScreen";
+import WalletNavigator from "./WalletNavigator";
+
+type RouteType = keyof typeof ROUTES;
+
+type RouteIconMap = { [key in RouteType]?: string };
+
+const ROUTE_ICON: RouteIconMap = {
+  MAIN_MESSAGES: "mail",
+  WALLET_HOME: "wallet",
+  DOCUMENTS_HOME: "document",
+  PREFERENCES_HOME: "cog",
+  MAIN_PROFILE: "user"
+};
+
+const getIcon = (routeName: RouteType): string => {
+  const unknownIcon = "???";
```

Comment:
no magic values, return an `Option<string>` or `string | undefined`

Reply:
btw if you use `ROUTES` keys as keys (e.g. `[ROUTES.MAIN_MESSAGES]`), `ROUTE_ICON[routeName]` should always be defined.